crates/red_knot_python_semantic/src/unpack.rs

```diff
@@ -54,8 +52,25 @@ impl<'db> Unpack<'db> {
     }
 
     /// Returns the range of the unpack target expression.
-    pub(crate) fn range(self, db: &'db dyn Db) -> TextRange {
-        self.target(db).range()
+    ///
+    /// `query_file` is the file for which the current query performs type inference.
+    /// It acts as a token of prove that we aren't accessing an AST node from a different file
+    /// than in which the current enclosing Salsa query (which would lead to cross-file dependencies).
+    #[inline]
+    pub(crate) fn range(self, db: &'db dyn Db, query_file: File) -> TextRange {
+        self.target(db, query_file).range()
+    }
+
+    /// The target expression that is being unpacked. For example, in `(a, b) = (1, 2)`, the target
+    /// expression is `(a, b)`.
+    ///
+    /// `query_file` is the file for which the current query performs type inference.
+    /// It acts as a token of prove that we aren't accessing an AST node from a different file
+    /// than in which the current enclosing Salsa query (which would lead to cross-file dependencies).
```

Comment:
```suggestion
    /// It acts as a token of proof that we aren't accessing an AST node from a different file
    /// to the one from which the current enclosing Salsa query is being called.
    /// Doing so would lead to cross-file dependencies, hurting incremental computation.
```